src/plugins/es_ui_shared/docs/docs/form_lib/examples/dynamic_fields.md

```diff
@@ -0,0 +1,278 @@
+---
+id: dynamic_fields
+title: Dynamic fields
+sidebar_label: Dynamic fields
+---
+
+## Basic
+
+Dynamic fields are fields that the user can add or remove in your form. Those fields will end up in an array of _values_ or an array of _objects_.
+
+Let's imagine a form that lets a user enter multiple parent / child relationships.
+
+```js
+export const DynamicFields = () => {
+  const fetchedData = {
+    name: 'My resource',
+    relationships: [
```

Comment:
Maybe not necessary, but I'm wondering if it would also be helpful to add a `gif` to demonstrate what dynamic fields in a form might look like/behave?

Reply:
Good point @cjcenizal, I will change the example but won't use `priority` but `subTitle` (priority would require a select component with options and I'd like to keep the JSX as lightweight as possible).

@alisonelizabeth That's a great idea. Let's see if we can get an MDX component (running react example inside the markdown) and if not we can use a gif as a replacement.